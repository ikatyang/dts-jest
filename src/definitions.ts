import * as ts from 'typescript';

export type TriggerDescription = null | string;

export interface ITrigger {
  line: number;
  kind: TriggerKind;
  description: TriggerDescription;
}

export interface ITriggerDescriptions {
  [line: number]: TriggerDescription;
}

export enum TriggerKind {
  None,
  Skip,
  Only,
  Show,
}

export interface ITSConfig {
  extends?: string;
  compilerOptions?: ts.CompilerOptions;
}

export type IRawTSConfig = undefined | string | ts.CompilerOptions;

export interface ISelfConfig {
  tsconfig?: IRawTSConfig;
  reporter?: string;
}

export const default_reporter_template = '\nInferred\n\n{{expression,2}}\n\nto be\n\n{{snapshot,2}}';

export interface IJestConfig {
  globals: {
    _dts_jest_: ISelfConfig;
  };
}

export const config_namespace = '_dts_jest_';
export const runtime_namespace = '_dts_jest_runtime_';

export const trigger_regex = /^\s*\/\/\s*@dts-jest\b(:\S*)?\s*(.+)?\s*$/;
export enum TriggerMatchIndex {
  Input,
  Flag,
  Description,
}
export type ITriggerMatchArray = [
  string,
  string | undefined,
  string | undefined
];

export interface IRuntimeSnapshotParameters {
  line: number;
}
export interface IRuntimeDescriptionParameters {
  line: number;
}
export interface IRuntimeReportParameters {
  line: number;
}

export interface IRuntimeMethods {
  report(parameters: IRuntimeReportParameters): string;
  snapshot(parameters: IRuntimeSnapshotParameters): string;
  description(parameters: IRuntimeDescriptionParameters): string;
}
