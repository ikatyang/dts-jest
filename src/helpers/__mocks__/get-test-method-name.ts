export const get_test_method_name = () => '[test-method-name]';
