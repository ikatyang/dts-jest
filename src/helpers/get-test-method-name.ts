import {TriggerKind} from '../definitions';

export const get_test_method_name = (kind: TriggerKind) => {
  switch (kind) {
    case TriggerKind.None:
    case TriggerKind.Show:
      return 'test';
    case TriggerKind.Only:
      return 'test.only';
    case TriggerKind.Skip:
      return 'test.skip';
    default:
      throw new Error(`Unexpected kind ${JSON.stringify(kind)}`);
  }
};
