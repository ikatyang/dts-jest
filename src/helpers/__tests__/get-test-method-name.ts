jest.unmock('../get-test-method-name');

import {TriggerKind} from '../../definitions';
import {get_test_method_name} from '../get-test-method-name';

it('should return correctly while kind is valid', () => {
  expect(get_test_method_name(TriggerKind.Show)).toMatchSnapshot('show');
  expect(get_test_method_name(TriggerKind.Skip)).toMatchSnapshot('skip');
  expect(get_test_method_name(TriggerKind.Only)).toMatchSnapshot('only');
  expect(get_test_method_name(TriggerKind.None)).toMatchSnapshot('none');
});

it('should throw error while flag is invalid', () => {
  expect(() => get_test_method_name(-1)).toThrowError();
});
