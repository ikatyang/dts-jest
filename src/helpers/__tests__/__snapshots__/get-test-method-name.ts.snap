// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`none 1`] = `"test"`;

exports[`only 1`] = `"test.only"`;

exports[`show 1`] = `"test"`;

exports[`skip 1`] = `"test.skip"`;
